find_email_addresses.py

```diff
@@ -1,59 +0,0 @@
-from bs4 import BeautifulSoup
-from collections import deque
-import requests
-import re
-import sys
-
-# Starting URL should be first argument from sys
-start_url = sys.argv[1]
-if not start_url.startswith("http://"):
-	start_url = "http://" + start_url 
-
-q = deque([start_url])
-
-urls_seen = set([])
-emails = set([])
-
-while q:
-	url = q.popleft()
-	urls_seen.add(url)
-
-	# Allow redirecting only on start_url
-	r = requests.get(url, allow_redirects = (url == start_url))
-
-	soup = BeautifulSoup(r.text, 'html.parser')
-
-	# Extract all email addresses and add them into the resulting set
-	new_emails = set(re.findall(r"[a-z0-9\.\-+_]+@[a-z0-9\.\-+_]+\.[a-z]+", r.text, re.I))
-	emails.update(new_emails)
-
-	# Iterate over all links found on the page
-	for link in soup.find_all('a'):
-
-		# It's possible to see 'a' elements without 'href'
-		try:		
-			new_url = link.attrs["href"]
-		except:
-			break
-
-		# Truncate extraneous slashes
-		if new_url.startswith("//"):
-			new_url = new_url[2:]
-
-		# Check if relative address is being used
-		elif new_url.startswith("/"):
-			new_url = start_url + new_url
-
-		if not new_url.startswith("http://"):
```

Comment:
For webpages using SSL, I'd need to somehow check that they require a secure connection and then use requests.get together with http.client.HTTPSConnection. As for cleaning up the URLs, I looked into urlparse.urlparse, but it doesn't seem to provide me with the requisite toolset